app/dashboard/views/custom_alias.py

```diff
@@ -250,6 +252,20 @@ def custom_alias():
     )
 
 
+def get_suffix(user: User) -> str:
```

Comment:
Hey it'd be best to make sure that the suffix isn't already used in other aliases as otherwise it'd increase the conflict chance :), for example when 2 users want to name their aliases `facebook.{random suffix}@alias.com`

Reply:
I wonder why this needs to be checked in advance? If a conflict does occur, it should be caught by the database when the alias is eventually created. To ensure that no existing suffix match, we'll need to go through all existing aliases in the database. The probability of a conflict is not very high in the first place, so it seems not very efficient/worth to check for conflict in advance every time.

Reply:
I think it's better that we can reduce the chance of conflict :). A check like `Alias.query.filter(Alias.email.endswith(suffix)).count()` might be enough to know whether a suffix has been used right? I think we can do something similar to this to look for a suffix that hasn't been used before https://github.com/simple-login/app/blob/ef2eb7f95958c8f9df8c38f594b3006331a3d915/app/models.py#L1111

Reply:
My point is that avoiding conflict at this stage is not necessary. It may increase unnecessary database queries and prolong alias creation time. I think each query against the suffix needs to search all the aliases in the entire alias table, and it's not an exact match - so it may not be possible to use indexes to speed it up.

Ultimately when adding an alias, the database should ensure that there are no two conflicting aliases, so in the most extreme case it will just throw an error and allow the user to reselect the suffix. Even this extreme case should be difficult to encounter, as the prefix-suffix namespace is large enough.

If you're still skeptical ...... maybe you can put a debug log to see what the probability of entering this 1000-times loop is with the current generation mechanism. Using a random suffix should result in a lower probability of conflict than the current word generator.

Reply:
Oh I see, let's keep this current approach and if too many users complain, we can add the check later :). The code makes sure that alias address is unique but displays an error if an alias already exists so this case is covered :).